src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -39,6 +46,8 @@ public class KestrelConfigurationLoader
         ConfigurationReader = new ConfigurationReader(configuration);
 
         _httpsConfigurationService = httpsConfigurationService;
+        _certificatePathWatcher = certificatePathWatcher;
+        Debug.Assert(!!reloadOnChange || (certificatePathWatcher is null), "If reloadOnChange is false, then certificatePathWatcher should be null");
```

Comment:
`!!`?

Reply:
I find `!a || b` the most readable way to express `a => b` and I thought `!!` made it clearer that this was such an expression.  I don't feel strongly about it.